test/shrine_test.rb

```diff
@@ -352,6 +352,17 @@ def respond_to?(name, include_private = false)
       assert_match /^[\w-]+$/, location
     end
 
+    it "gets extension from shrine-url-style id with query params" do
+      uploaded_file = @uploader.class.const_get("UploadedFile").new("id" => "http://example.com/path.html?key=value", "storage" => "cache")
+      location = @uploader.generate_location(uploaded_file)
+      assert_match /\.html$/, location
+
+      uploaded_file = @uploader.class.const_get("UploadedFile").new("id" => "http://example.com/path?key=value", "storage" => "cache")
+      location = @uploader.generate_location(uploaded_file)
+      refute_match /key=value/, location
+      assert_match /^[\w-]+$/, location
+    end
+
```

Comment:
We should probably also make a unit test for `UploadedFile#extension` in `test/uploaded_file_test.rb`.

Reply:
I can add that.

Reply:
done, thanks, didn't notice those tests!